src/common/hooks/useRestoreToken.tsx

```diff
@@ -44,34 +44,66 @@ export const useRestoreToken = (
+      contractInstance.address,
+      previousBeneficiary,
+      previousHolder
+    );
+
+    const escrowDeploymentTx = await escrowDeploymentReceipt.wait();
+    const deployedTitleEscrowArgs = escrowDeploymentTx.events?.find((event) => event.event === "TitleEscrowDeployed")
+      ?.args;
+    if (!deployedTitleEscrowArgs || !deployedTitleEscrowArgs[0])
+      throw new Error(`Address for deployed title escrow cannot be found. Tx: ${JSON.stringify(escrowDeploymentTx)}`);
+    const deployedTitleEscrowAddress = deployedTitleEscrowArgs[0];
+
+    // use minter restore token method to send token back to last known bene and holder
+    const sendTokenReceipt = await contractInstance.sendToken(deployedTitleEscrowAddress, tokenId);
+    const sendTokenTx = await sendTokenReceipt.wait();
+    const sendTokenArgs = sendTokenTx.events?.find((event) => event.event === "Transfer")?.args;
+    if (!sendTokenArgs || sendTokenArgs[1] !== deployedTitleEscrowAddress)
+      throw new Error(`Token was not restored to owner and beneficiary. Tx: ${JSON.stringify(sendTokenTx)}`);
+  };
+
   const restoreToken = async (previousBeneficiary: string, previousHolder: string): Promise<void> => {
     setState("INITIALIZED");
     try {
-      if (!tokenId) throw new Error("Ownership data is not provided");
-      const titleEscrowCreatorContract = await getTitleEscrowCreator(provider as providers.Provider);
-
-      // Deploy new title escrow smart contract to own document
-      if (!contractInstance?.address) throw new Error("Token Registry Instance should have address");
-      setState("PENDING_CONFIRMATION");
-      const escrowDeploymentReceipt = await titleEscrowCreatorContract.deployNewTitleEscrow(
-        contractInstance.address,
-        previousBeneficiary,
-        previousHolder
-      );
-
-      const escrowDeploymentTx = await escrowDeploymentReceipt.wait();
-      const deployedTitleEscrowArgs = escrowDeploymentTx.events?.find((event) => event.event === "TitleEscrowDeployed")
-        ?.args;
-      if (!deployedTitleEscrowArgs || !deployedTitleEscrowArgs[0])
-        throw new Error(`Address for deployed title escrow cannot be found. Tx: ${JSON.stringify(escrowDeploymentTx)}`);
-      const deployedTitleEscrowAddress = deployedTitleEscrowArgs[0];
-
-      // use minter restore token method to send token back to last known bene and holder
-      const sendTokenReceipt = await contractInstance.sendToken(deployedTitleEscrowAddress, tokenId);
-      const sendTokenTx = await sendTokenReceipt.wait();
-      const sendTokenArgs = sendTokenTx.events?.find((event) => event.event === "Transfer")?.args;
-      if (!sendTokenArgs || sendTokenArgs[1] !== deployedTitleEscrowAddress)
-        throw new Error(`Token was not restored to owner and beneficiary. Tx: ${JSON.stringify(sendTokenTx)}`);
+      const isLatestTradeTrustErc721 = await contractInstance?.supportsInterface("0x9f9e69f3");
```

Comment:
`isLatestTradeTrustErc721` tells the user nothing about the code. Instead, if you write `supportsSendToTitleEscrow` it will show the intent clearer.

Reply:
ok have changed it.